src/system/DurationTimer.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <string>
+
+#include <system/SystemConfig.h>
+
+/**
+ * @file
+ * 
+ *  Uses SystemClock.h HAVE_CLOCK_GETTIME to determine board type i.e. esp32 or nRF etc.
+ *  CHIP_DEVICE_USES_SYS_TIME flag indicates nRF board whose clock implementation is available in Syste,Clock.h.
+ *  CHIP_DEVICE_USES_TIME_H flag for esp32 uses a time.h timespec
```

Comment:
This is not useful API documentation.  Please document what this class is meant for.

Is it just to measure durations?  Is it to measure intervals and allow correlating them with wall-clock times?  Something else?